engines/ultima/ultima8/world/current_map.cpp

```diff
@@ -1200,7 +1200,7 @@ bool CurrentMap::sweepTest(const Point3 &start, const Point3 &end,
 					}
 
 					// Now add it
-					sw_it = hit->insert(sw_it, SweepItem(other_item->getObjId(), first, last, touch, touch_floor, blocking, dirs));
+					hit->insert(sw_it, SweepItem(other_item->getObjId(), first, last, touch, touch_floor, blocking, dirs));
```

Comment:
Why are you removing the return value here? It's used throughout the for loop

Reply:
The previous implementation in the Ultima engine incorrectly returned the iterator passed as a parameter instead of the one for the newly added element, which means that this needed to be changed so that it could be handled properly in common code.

Reply:
I think this new logic is correct, headsup to @OMGPizzaGuy just in case my reasoning is wrong.

Reply:
It looks like the iterator is also only being retained for an optimization that has guards around it.